src/engine/QueryExecutionTree.cpp

```diff
@@ -91,22 +92,31 @@ void QueryExecutionTree::setVariableColumns(
 // _____________________________________________________________________________
 template <QueryExecutionTree::ExportSubFormat format>
 ad_utility::stream_generator::stream_generator
-QueryExecutionTree::generateResults(const vector<string>& selectVars,
+QueryExecutionTree::generateResults(const std::variant<vector<string>,char> & selectVarsOrAsterisk,
                                     size_t limit, size_t offset) const {
   // They may trigger computation (but does not have to).
   shared_ptr<const ResultTable> resultTable = getResult();
   LOG(DEBUG) << "Resolving strings for finished binary result...\n";
   vector<std::optional<pair<size_t, ResultTable::ResultType>>> validIndices;
-  for (auto var : selectVars) {
-    if (ad_utility::startsWith(var, "TEXT(")) {
-      var = var.substr(5, var.rfind(')') - 5);
-    }
-    auto it = getVariableColumns().find(var);
-    if (it != getVariableColumns().end()) {
-      validIndices.push_back(pair<size_t, ResultTable::ResultType>(
+  if(std::holds_alternative<char>(selectVarsOrAsterisk)) {
+    auto it = getVariableColumns().begin();
+    while (it != getVariableColumns().end()) {
+      validIndices.emplace_back(pair<size_t, ResultTable::ResultType>(
```

Comment:
This should be a range-based for loop:
`for (const auto& element : getVariableColumns()`
And by the way, this is an infinite loop